Opting in the container based Travis builds

bsipocz opened the pull request:
This is to address #1494. 
However installing the ``openjpeg`` dependency still requires ``sudo`` as that is not in the travis apt package list. Thus this needs a workaround.

Cadair commented:
the information in #1498 could be used to work around the sudo requirement.

Cadair commented:
I am going to have to get @dpshelio to buy you some kind of beverage at EiroSciPy!

Let me know if you have any issues with this one.

bsipocz commented:
@Cadair - NP,  thanks. 
With this latest rebase I've removed the temporary debug stuff and switched back all the builds. This openjpeg dependency wasn't trivial. But now it should be good to go. Sorry for overloading travis.

Cadair commented:
Hey! The doctests ran! It didn't pass though.

bsipocz commented:
Yes, I saw them during an earlier run, too.

Cadair commented:
Still that's not an issue about this. 

:+1:

bsipocz commented:
Another point for this PR  though is that now the container based builds are running, but I can't see that they are particularly faster than the legacy system (but they were much faster for astropy and photutils, where we test on more setups.)

Cadair commented:
Did you notice a speedup in execution time? I think they are supposed to start faster.

bsipocz commented:
We did definitely see a speedup with photutils, and also with the core astropy. Also they started up much faster than before. Here I can't really see a significant difference, but the scatter in the total exec time is somehow quite big for sunpy. 
 It's a fast build (41-43 mins) but there were some fast builds (~48-50mins) before, too however also there were long ones too (>1h). They say that the exec time in the container based system should be less sensitive for loads of other projects, so it sounds to be worth to switch.

dpshelio commented:
I love doctests!!!  Great!!! 
@bsipocz be sure that I will give you one drink at EuroScipy from me on top of that from @Cadair

dpshelio commented:
Oh yes, :+1:
